src/main/java/org/rumbledb/runtime/functions/input/UrlValidator.java

```diff
@@ -16,28 +16,34 @@
 
     private static final String[] allowedSchemes = { "file", "hdfs", "s3", "s3a", "s3n", "wasb", "gs", "root" };
 
-    public static boolean exists(String url) {
+    public static boolean exists(String url, ExceptionMetadata metadata) {
         URI locator = null;
         try {
             locator = new URI(url);
         } catch (URISyntaxException e) {
-            throw new OurBadException("Malformed URL: " + url);
+            throw new CannotRetrieveResourceException("Malformed URL: " + url, metadata);
         }
-        if (
-            !locator.isAbsolute()
-                || Arrays.asList(allowedSchemes).contains(locator.getScheme())
-        ) {
-            JavaSparkContext sparkContext = SparkSessionManager.getInstance().getJavaSparkContext();
+        if (locator.isAbsolute() && !Arrays.asList(allowedSchemes).contains(locator.getScheme())) {
+            throw new OurBadException("Cannot interpret URL: " + url);
+        }
+        {
             try {
-                FileSystem fileSystem = FileSystem.get(sparkContext.hadoopConfiguration());
+                FileContext fileContext = FileContext.getFileContext();
                 Path path = new Path(url);
-                return url.contains("*") || fileSystem.exists(path);
+                return url.contains("*") || fileContext.util().exists(path);
+
+            } catch (UnsupportedFileSystemException e) {
+                throw new CannotRetrieveResourceException(
+                        "No file system is configured for scheme " + url + "!",
+                        metadata
+                );
             } catch (IOException e) {
-                throw new SparksoniqRuntimeException("Error while accessing hadoop filesystem.");
+                e.printStackTrace();
```

Comment:
Is this printStackTrace intentionally kept here?

Reply:
Yes, to debug more easily if something happens.